calico/felix/plugins/fiptgenerator.py

```diff
@@ -576,13 +578,29 @@ def drop_rules(self, ip_version, chain_name, rule_spec=None, comment=None,
                 "Invalid comment %r" % comment
             comment_str = '-m comment --comment "%s"' % comment
 
-        parts = [p for p in [ipt_action,
-                             chain_name,
-                             rule_spec,
-                             '--jump DROP',
-                             comment_str]
-                 if p is not None]
-        return [' '.join(parts)]
+        rules = []
+
+        if self.ACTION_ON_DROP.startswith("LOG-"):
```

Comment:
I'm amused that this doesn't use the on_deny parameter, passed in to rule_to_iptables_fragments.